packages/pwa-kit-dev/src/configs/webpack/config.js

```diff
@@ -556,7 +556,25 @@ const requestProcessor =
         })
         .build()
 
-module.exports = [client, ssr, renderer, clientOptional, requestProcessor]
+// TODO: can we make this one config with multiple entries? would that dedupe the dependencies?
```

Comment:
You could possibly use multiple entries kinda like this:

```
const extensionEntries = extensions.map(...)

// config
entry: extensionEntries,
output: {
    path: `${buildDir}/extensions/[name]`,
    filename: 'setup-server.js'
}
```

I'm not sure what the effect is on dependencies, but I guess I'm also not sure exactly what you mean by having duplicate dependancies. Maybe you can explain what is happening a little more.

Reply:
I've been thinking about this and I have a new perspective that makes me want to create separate builds.

If we build all extensions into a single build, then at runtime, there could be chances where 1 bad extension could mess up with all extensions.

If we build one file per extension, then we isolate the problem, if there is a bad extension, other extensions can still be applied.

Reply:
I thought that we were always creating a separate output per extensions? Was that not the case in your implementation?